File: unity/Assets/Scripts/Multiplayer/Rooms/Updates/UpdatePlayersRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Michitai.Multiplayer.Rooms.Updates
{
    /// <summary>
    /// Internal request data for sending player updates in Unity.
    /// Uses serialized JSON string for data.
    /// </summary>
    [System.Serializable]
    internal class UpdatePlayersRequest
    {
        /// <summary>
        /// The target players for the update (all, host, others, specific).
        /// </summary>
        public string target_players = ERoomTargetPlayers.All.ToString().ToLower();

        /// <summary>
        /// Specific player IDs if target_players is specific.
        /// </summary>
        public int[] target_players_ids;

        /// <summary>
        /// The type of update being sent.
        /// </summary>
        public string type;

        /// <summary>
        /// Serialized JSON string of the update data (Unity mode).
        /// </summary>
        public string data_json;

        /// <summary>
        /// Initializes a new UpdatePlayersRequest.
        /// </summary>
        /// <param name="targetPlayers">The target players for the update.</param>
        /// <param name="type">The type of update being sent.</param>
        /// <param name="dataJson">Serialized JSON string of the update data.</param>
        /// <param name="targetPlayerIds">Specific player IDs if target_players is specific.</param>
        public UpdatePlayersRequest(ERoomTargetPlayers targetPlayers, string type, string dataJson = null, int[] targetPlayerIds = null)
        {
            this.target_players = targetPlayers.ToString().ToLower();
            this.target_players_ids = targetPlayerIds;
            this.type = type;
            this.data_json = dataJson;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/rooms/updates/UpdatePlayersRequest.java
package com.michitai.multiplayer.rooms.updates;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.michitai.multiplayer.rooms.actions.ERoomTargetPlayers;

/**
 * Request for sending updates to target players.
 */
public class UpdatePlayersRequest {
    @JsonProperty("target_players")
    private String targetPlayers;

    @JsonProperty("type")
    private String type;

    @JsonProperty("data")
    private String data;

    @JsonProperty("target_players_ids")
    private int[] targetPlayersIds;

    public UpdatePlayersRequest(ERoomTargetPlayers targetPlayers, String type, String data, int[] targetPlayersIds) {
        this.targetPlayers = targetPlayers.name().toLowerCase();
        this.type = type;
        this.data = data;
        this.targetPlayersIds = targetPlayersIds;
    }

    public String getTargetPlayers() {
        return targetPlayers;
    }

    public void setTargetPlayers(String targetPlayers) {
        this.targetPlayers = targetPlayers;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getData() {
        return data;
    }

    public void setData(String data) {
        this.data = data;
    }

    public int[] getTargetPlayersIds() {
        return targetPlayersIds;
    }

    public void setTargetPlayersIds(int[] targetPlayersIds) {
        this.targetPlayersIds = targetPlayersIds;
    }
}
